streams/src/main/java/org/apache/kafka/streams/state/internals/PrefixedSessionKeySchemas.java

```diff
@@ -204,7 +203,8 @@ public static void writeBinary(final ByteBuffer buf,
                                        final long endTime) {
             buf.putLong(endTime);
             buf.putLong(startTime);
-            buf.put(key.get());
+            if (key != null)
```

Comment:
Can `key` ever be `null` here? (nit: add `{}` to block)

Reply:
Yes it's possible since it's used to write lower/upper boundaries in which keys could be null.